### PR TITLE
Don't flush writer if unused

### PR DESCRIPTION
**Pull request by rikonor**

### CODE REVIEW

File: container/writer.go [hunk 1]
```diff
@@ -96,6 +96,10 @@ func (avroWriter *Writer) WriteRecord(record AvroRecord) error {
 }
 
 func (avroWriter *Writer) Flush() error {
+	if !avroWriter.headerWritten {
+		return nil
+	}
```
Comment (Seanstoppable) [inline container/writer.go]: We do things like close and reset the writer below. Do we not need to do that still?

Comment (rikonor) [reply container/writer.go]: Ah yes, my bad.

It looks like I'd be able to put that `if` statement right before `block ...`, does that seem right? That way we close but not necessarily continue writing.

File: container/writer.go [hunk 2]
```diff
@@ -103,6 +103,11 @@ func (avroWriter *Writer) Flush() error {
 		fwWriter.Reset(avroWriter.blockBuffer)
 	}
 
+	// dont flush if unused
+	if !avroWriter.headerWritten {
+		return nil
+	}
+
```
Comment (Seanstoppable) [inline container/writer.go]: Farther down we have 

avroWriter.blockBuffer.Reset()
avroWriter.nextBlockRecords = 0

I don't know if they are necessary.

Perhaps we just wrap the serialization in
``` 
if avroWriter.headerWritten {
}
```

Comment (rikonor) [reply container/writer.go]: Yep, that's fine by me. Changed.

**Review by saurori — APPROVED** [review]

**Review by Seanstoppable — APPROVED** [review]

**Pull request merged**